src/Lykke.blue.Service.ReferralLinks/Controllers/ReferralLinksController.cs

```diff
@@ -0,0 +1,480 @@
+
+            return Ok(referraLinksStatistics);
+        }
+
+        /// <summary>
+        /// Request money transfer referral link - reserved for version 2   
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        //[HttpPost("giftCoinLinks")]
+        //[SwaggerOperation("RequestGiftCoinsReferralLink")]
+        //[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
+        //[ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.NotFound)]
+        //[ProducesResponseType(typeof(RequestRefLinkResponse), (int)HttpStatusCode.Created)]        
+        private async Task<IActionResult> RequestGiftCoinsReferralLink([FromBody] GiftCoinsReferralLinkRequest request)
+        {
+            if (request == null)
+            {
+                return await LogAndReturnBadRequest("", ControllerContext, Phrases.InvalidRequest);
+            }
+
+            if (request.Amount <= 0)
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidAmount);
+            }
+
+            if (String.IsNullOrEmpty(request.SenderClientId))
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidSenderClientId);
+            }
+
+            var asset = (await _assets.GetDictionaryAsync()).Values.FirstOrDefault(v => v.DisplayId == request.Asset);
+
+            if (String.IsNullOrEmpty(request.Asset) || asset == null)
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidAsset);
+            }
+
+            var clientBalances = await _balancesClient.GetClientBalances(request.SenderClientId);
+
+            if (clientBalances == null)
+            {
+                return await LogAndReturnNotFound(request, ControllerContext, $"Cant get clientBalance of asset {asset.Symbol} for client id {request.SenderClientId} from service {_settings.BalancesServiceClient.ServiceUrl}");
+            }
+
+            var balance = clientBalances.FirstOrDefault(x => x.AssetId == asset.Id)?.Balance;
+
+            if (!balance.HasValue || balance.Value < request.Amount)
+            {
+                return await LogAndReturnBadRequest(request, ControllerContext, Phrases.InvalidTreesAmount);
+            }
+
+            var referralLink = await _referralLinksService.CreateGiftCoinsLink(request);
+
+            await LogInfo(request, ControllerContext, referralLink.ToJson());
+
+            return Created(uri: $"api/referralLinks/{referralLink.Id}", value: new RequestRefLinkResponse { RefLinkId = referralLink.Id, RefLinkUrl = referralLink.Url });
+        }
+
+
+        /// <summary>
+        /// Request invitation referral link.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost("invitationLinks")]
```

Comment:
This can be changed into [HttpPost("invitation")]. In this way we will have nice-looking method POST "api/referralLinks/invitation"

Reply:
Changed